src/components/pages/books/Books.jsx

```diff
@@ -1,19 +1,27 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
-import { removeBook } from '../../../redux/books/booksSlice';
+import { removeBook, fetchData } from '../../../redux/books/booksSlice';
 import './books.css';
-
 import AddNewBook from './addNewBook';
 
 const Books = () => {
   const dispatch = useDispatch();
-  const books = useSelector((state) => state.bookstore.books);
+  const books = useSelector((state) => state.bookstore);
+
+  useEffect(() => {
+    dispatch(fetchData());
+  }, [dispatch]);
+  const bookLists = books.booklist.map((book, index) => ({
+    ...book,
+    item_id: book.item_id,
+    key: `${book.item_id}-${index}`,
+  }));
 
   return (
     <div className="book">
       <div className="book-section">
-        {books.map((book) => (
-          <div key={book.item_id} className="book-section-one">
+        {bookLists.map((book) => (
+          <div key={book.key} className="book-section-one">
             <div className="action-section">
               <span>{book.category}</span>
               <div className="book-title">
```

Comment:
- [ ] I noticed that when I add a book, the details of the added book are not displayed until I refresh the page. It shows empty placeholders till I refresh the page. Kindly fix this and test that the book shows after it is added. 👍 💻 

![chrome-capture-2023-5-21](https://github.com/AleWaweru/Bookstore/assets/77106962/afcb9407-2b50-441c-a986-db5d45cdf87f)

Reply:
done.